scene/gui/color_picker.cpp

```diff
@@ -1531,6 +1652,68 @@ void ColorPicker::_pick_finished() {
 	picker_window->hide();
 }
 
+void ColorPicker::_update_menu_items() {
+	if (!options_menu) {
+		options_menu = memnew(PopupMenu);
+		add_child(options_menu);
+		options_menu->connect("id_pressed", callable_mp(this, &ColorPicker::_options_menu_cbk));
+	}
+
+	options_menu->clear();
+	options_menu->reset_size();
+
+	if (preset_cache.size()) {
+		options_menu->add_icon_item(get_theme_icon(SNAME("save"), SNAME("FileDialog")), RTR("Save"), static_cast<int>(MenuOption::MENU_SAVE));
+		options_menu->set_item_tooltip(-1, ETR("Save the current color palette to reuse later."));
+	}
+	options_menu->add_icon_item(get_theme_icon(SNAME("load"), SNAME("FileDialog")), RTR("Load"), static_cast<int>(MenuOption::MENU_LOAD));
+	options_menu->set_item_tooltip(-1, ETR("Load existing color palette."));
+
+	if (Engine::get_singleton()->is_editor_hint()) {
+		options_menu->add_icon_item(get_theme_icon(SNAME("load"), SNAME("FileDialog")), RTR("Quick Load"), static_cast<int>(MenuOption::MENU_QUICKLOAD));
+		options_menu->set_item_tooltip(-1, ETR("Load existing color palette."));
+	}
+}
+
+void ColorPicker::_update_menu() {
+	_update_menu_items();
+	Rect2 gt = menu_btn->get_screen_rect();
+	menu_btn->reset_size();
+	int min_size = menu_btn->get_minimum_size().width;
+	Vector2 popup_pos = gt.get_end() - Vector2(min_size, 0);
+	options_menu->set_position(popup_pos);
+	options_menu->popup();
+}
+
+void ColorPicker::_options_menu_cbk(int p_which) {
+	if (!file_dialog) {
+		file_dialog = memnew(FileDialog);
+		add_child(file_dialog, false, INTERNAL_MODE_FRONT);
```

Comment:
```suggestion
		add_child(file_dialog, false, INTERNAL_MODE_FRONT);
		file_dialog->force_parent_owned();
```

Reply:
Why do I only see it being used in `TileMapLayer`? What about other places? Why don't other places use it?